src/lib.rs

```diff
@@ -239,14 +258,26 @@ fn setup_bitcoind(
         log::info!("Creating a new watchonly wallet on bitcoind.");
         bitcoind.create_watchonly_wallet(&config.main_descriptor)?;
         log::info!("Watchonly wallet created.");
-    } else {
+    } else if !is_remote {
         #[cfg(windows)]
         if !cfg!(test) && !wo_path.exists() {
             return Err(StartupError::NoWatchonlyInDatadir);
         }
     }
     log::info!("Loading our watchonly wallet on bitcoind.");
-    bitcoind.maybe_load_watchonly_wallet()?;
+    if is_remote {
```

Comment:
Could you add this `if is_remote {}` as an `else {}` to the `if fresh_data_dir {} else if !is_remote {}` above? I think we only need to check for a legacy wallet if it's not a fresh data dir, as otherwise we will have just created a wallet with the new name.

Reply:
Yes. Even clearer:
- If fresh data dir
- Else if remote
- Else (left untouched)

Reply:
does `create_watchonly_wallet()` also load the wallet in bitcoind? if so why we were calling `maybe_load_watchonly_wallet()` prior to this change?

Reply:
Yep, it loads the wallet also. I suppose in that case the call to `maybe_load_watchonly_wallet()` could have been in the `else` block only, but calling it on an already-loaded wallet is also fine as it returns immediately.